### ISSUE TITLE
Support for azurerm_dns_zone modify soa_record

### ISSUE THREAD

**Issue by dpa456** [hashicorp/terraform-provider-azurerm#14597]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

I want to change the ttl value of soa using azurerm_dns_zone when create/update.

However, azurem_dns_zone requires host_name.

Basically, I know that when dns zone is created, it is designated as a default value, but is it necessary to specify this value again?

### New or Affected Resource(s)

azurerm_dns_zone

### Potential Terraform Configuration

resource "azurerm_dns_zone" "example-public" {
  name                = "mydomain.com"
  resource_group_name = azurerm_resource_group.example.name
 
  soa_record {
     ttl = 3000
  }
}

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/dns_zone#soa_record

**Comment by tombuildsstuff** [comment]
hi @dpa456 

Thanks for opening this issue.

Taking a look through this is an issue about Terraform Configuration rather than a bug in the Azure Provider - you can both set this property at creation and modify it during the update via `terraform apply`.

This forum is intended to be used for feature enhancements and bugs in the Azure Provider - so that we can keep this forum focused on that we instead ask that broader questions are raised [using one of the Community Resources](https://www.terraform.io/community.html). As such I'm going to close this issue for the moment, but I believe you should be able to get an answer for this [using one of the Community Resources](https://www.terraform.io/community.html).

Thanks!

**Comment by dpa456** [comment]
Thank you for your quick confirmation.
